src/components/user/ha-users-picker.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  customElement,
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import { fireEvent } from "../../common/dom/fire_event";
+import type { PolymerChangedEvent } from "../../polymer-types";
+import type { HomeAssistant } from "../../types";
+import { User } from "../../data/user";
+import "./ha-user-picker";
+
+@customElement("ha-users-picker")
+class HaUsersPickerLight extends LitElement {
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
+  @property() public value?: string[];
+
+  @property({ attribute: "picked-user-label" })
+  public pickedUserLabel?: string;
+
+  @property({ attribute: "pick-user-label" })
+  public pickUserLabel?: string;
+
+  @property({ type: Array, attribute: "users" })
+  public users?: User[];
+
+  protected render(): TemplateResult {
+    if (!this.hass) {
+      return html``;
+    }
+
+    const currentUsers = this._currentUsers;
+    return html`
+      ${currentUsers.map(
+        (user_id) => html`
+          <div>
+            <ha-user-picker
+              .label=${this.pickedUserLabel}
+              .curValue=${user_id}
+              .hass=${this.hass}
+              .value=${user_id}
+              .users=${this.users}
+              @value-changed=${this._userChanged}
+            ></ha-user-picker>
+          </div>
+        `
+      )}
+      <ha-user-picker
+        .label=${this.pickUserLabel}
+        .hass=${this.hass}
+        .users=${this.users}
+        @value-changed=${this._addUser}
+      ></ha-user-picker>
+    `;
+  }
+
+  private get _currentUsers() {
+    return this.value || [];
+  }
+
+  private async _updateUsers(users) {
+    this.value = users;
+    fireEvent(this, "value-changed", {
+      value: users,
+    });
+  }
+
+  private _userChanged(event: PolymerChangedEvent<string>) {
+    event.stopPropagation();
+    const curValue = (event.currentTarget as any).curValue;
+    const newValue = event.detail.value;
+    if (newValue === curValue) {
+      return;
+    }
+    if (newValue === "") {
+      this._updateUsers(this._currentUsers.filter((user) => user !== curValue));
+    } else {
+      this._updateUsers(
+        this._currentUsers.map((user) => (user === curValue ? newValue : user))
+      );
+    }
+  }
+
+  private async _addUser(event: PolymerChangedEvent<string>) {
+    event.stopPropagation();
+    const toAdd = event.detail.value;
+    (event.currentTarget as any).value = "";
```

Comment:
@bramkragten this doesn't work, because `ha-user-picker` is doing this:
```typescript
const newValue = ev.detail.item.dataset.userId;

if (newValue !== this._value) {
```
And changing the value doesn't change the selected item. What's the best way to resolve this?

Reply:
That's actually weird, because it's comparing to `ev.detail.item.dataset.userId`, yet setting to `ev.detail.value`:
```typescript
private _userChanged(ev) {
    const newValue = ev.detail.item.dataset.userId;

    if (newValue !== this._value) {
      this.value = ev.detail.value;
      setTimeout(() => {
        fireEvent(this, "value-changed", { value: newValue });
        fireEvent(this, "change");
      }, 0);
    }
  }
```

Reply:
Yeah, that doesn't seem correct. I think newValue should be `ev.detail.value`?

Reply:
This is really strange. So `ev.detail.value` doesn't even exist (which means that `this.value = ev.detail.value;` is probably wrong), and `ev.newValue` is always `undefined`.
It seems like `ev.detail.item.dataset.userId` is indeed the right way to grab the currently selected user, but it's unclear how to programmatically select a user...

Reply:
This is a red herring - this method on `ha-user-picker` isn't even called by setting value to `""`, and it's unclear that `value` even has any meaning (I thought it was called, because it was called twice when picking a user, so I assumed one of those was a result of setting value to `""`).
I'm completely stumped 🤷‍♂️

Reply:
some more investigation - the reason it's called twice:
1. Once for the "add user" box
2. Once for the new box that's created to represent the new value

I still don't understand how to set the value of the "add user" box to "No user" again.

Reply:
hmm, looking into [the source for this component](https://github.com/PolymerElements/paper-dropdown-menu/blob/f98f0a1500dd8f5e329137be696b37f5934088f5/paper-dropdown-menu-light.js):
```
/**
     * The value for this element that will be used when submitting in
     * a form. It reflects the value of `selectedItemLabel`. If set directly,
     * it will not update the `selectedItemLabel` value.
     */
```

Both `selectedItem` and `selectedItemLabel` are read only, so there's no way to set them...

Reply:
You should be able to set the `value` to undefined or `""`, as `selected` would then change to the first item.

Reply:
I tried both and neither worked. The doc I pasted above also says that changing `value` does not change the `selectedItemLabel`, but there are no instructions on how to actually change it.

Reply:
I'll put this PR on my todo list :-)

Reply:
thanks

Reply:
Any chance we can get it in time for the beta? The core feature is already merged

Reply:
I'll try 🙈

Reply:
Should be fixed now